core/src/main/java/com/datastax/oss/driver/api/core/DriverInfo.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core;
+
+/** Exposes information about the driver. */
+public interface DriverInfo {
+
+  /** @return the driver full name. */
+  String getName();
```

Comment:
Not sure if this is really needed, it doesn't bring any additional info than the artifactId.

On the other hand, it would be nice to know if the driver is shaded. We can't do it through `Driver.properties` because it's generated at compile time, but we could do a runtime check on one of the Netty classes.